spec/features/self_consent_gillick_competent_spec.rb

```diff
@@ -47,17 +47,41 @@ def and_there_is_a_child_without_parental_consent
   def when_the_nurse_assesses_the_child_as_gillick_competent
     click_on @child.full_name
 
-    click_on "Assess Gillick competence"
-    click_on "Give your assessment"
+    click_link "Give your assessment"
+    click_button "Give your assessment"
+
+    # try submitting without filling in the form
+    click_on "Continue"
+    expect(page).to have_content("There is a problem")
+    expect(page).to have_content("Choose if they are Gillick competent")
 
     choose "Yes, they are Gillick competent"
+    click_on "Continue"
 
-    fill_in "Give details of your assessment",
+    # try submitting without filling in the form
+    click_on "Continue"
+    expect(page).to have_content("Enter details of your assessment")
+
+    fill_in "Details of your assessment",
             with: "They understand the benefits and risks of the vaccine"
     click_on "Continue"
+
+    expect(page).to have_content("Check and confirm")
+    expect(page).to have_content(
+      ["Are they Gillick competent?", "Yes, they are Gillick competent"].join
+    )
+    expect(page).to have_content(
+      [
+        "Details of your assessment",
+        "They understand the benefits and risks of the vaccine"
+      ].join
+    )
+    click_on "Save changes"
   end
```

Comment:
This could probably be broken up into multiple steps, non-blocking though

Reply:
Yes, it is getting a bit long... split up here [601c3dd](https://github.com/nhsuk/manage-vaccinations-in-schools/pull/1368/commits/601c3dd58e92732c891119891ed4e9581f1fc839)